cluster-autoscaler/core/scaledown/actuation/actuator_test.go

```diff
@@ -1150,15 +1144,21 @@ func generatePods(count int, prefix string) []*apiv1.Pod {
 		if prefix != "" {
 			name = prefix + "-" + name
 		}
-		result = append(result, generatePod(name))
+		result = append(result, generatePod(name, prefix))
 	}
 	return result
 }
 
-func generatePod(name string) *apiv1.Pod {
+func generatePod(name string, node string) *apiv1.Pod {
 	return &apiv1.Pod{
-		ObjectMeta: metav1.ObjectMeta{Name: name},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+			Annotations: map[string]string{
+				"cluster-autoscaler.kubernetes.io/safe-to-evict": "true",
```

Comment:
Why add  this annotation on all test pods? Doesn't think reduce scope of the testing?

Reply:
A pod that is not replicated and doesn't have a safe-to-evict annotation will block deletion when checking in GetPodsForDeletionOnNodeDrain. https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/utils/drain/drain.go#L211